terraformInstall.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+## CONDITIONAL 1
+echo "Installing prerequisites..."
+sudo apt-get update && sudo apt-get install -y gnupg software-properties-common
+
+#!/bin/bash
+
+# Check if gpg is installed
+if ( which gpg > /dev/null 2>&1 )
+then
+    echo "gpg is installed."
+else
+    echo "Error: gpg is not installed. Please install it before running this script."
+    exit 1
+fi
+
+# Check if the keyring file exists using stat
+if (stat /usr/share/keyrings/hashicorp-archive-keyring.gpg > /dev/null 2>&1)
+then
+    echo "HashiCorp GPG key already exists."
+else
+    echo "HashiCorp GPG key not found. Adding HashiCorp GPG key."
+
+    # Add the HashiCorp GPG key
+    wget -O- https://apt.releases.hashicorp.com/gpg |
+    gpg --dearmor |
```

Comment:
you may have weird line breaks in here, keep this all on one line.

Reply:
This is copied from the hashicorp site 
<img width="1680" alt="Screenshot 2024-11-18 at 11 03 44 AM" src="https://github.com/user-attachments/assets/28bf9bc2-12f8-4db
[](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli)